Tools/clinic/clinic.py

```diff
@@ -4428,7 +4426,7 @@ class DSLParser:
     keyword_only: bool
     positional_only: bool
     group: int
-    parameter_state: int
+    parameter_state: ParamState
```

Comment:
This isn't related to the `truthy-bool` check, but it's a tiny typing-related fix that doesn't feel like it deserves its own PR. Mypy doesn't flag it (since it's not technically incorrect, it could just be slightly more specific), but I noticed pyright flagged it when I was editing `clinic.py` in VSCode the other day.